### PR TITLE
Add example to Interferometer

### PR DESCRIPTION
**Pull request by ankit27kh**
**Description of the Change:**
add documentation example to Interferometer
**Benefits:**
better documentation
**Possible Drawbacks:**
none
**Related GitHub Issues:**
#1727 #1735

### CODE REVIEW

**Comment by ankit27kh** [comment]
Hi @antalszava, this is the last template with no example. With this we can close both of the related issues.

**Comment by ankit27kh** [comment]
Also, Interferometer does not have a `shape` method. As it requires different parameters with different shapes, there should be this method. Should I make a PR for it?

File: pennylane/templates/subroutines/interferometer.py [hunk 1]
```diff
@@ -132,6 +132,56 @@ def Interferometer(theta, phi, varphi, wires, mesh="rectangular", beamsplitter="
 
     Raises:
         ValueError: if inputs do not have the correct format
+
+    Example:
+
+        The template requires :math:`3` sets of parameters. The ``mesh`` and ``beamsplitter`` arguments are optional and
```
Comment (Jaybsoni) [inline pennylane/templates/subroutines/interferometer.py:138]:
```suggestion
        The template requires :math:`3` sets of parameters. The ``mesh`` and ``beamsplitter`` keyword arguments are optional and
```

File: pennylane/templates/subroutines/interferometer.py [hunk 2]
```diff
@@ -132,6 +132,56 @@ def Interferometer(theta, phi, varphi, wires, mesh="rectangular", beamsplitter="
...
+                return qml.expval(qml.Identity(0))
+
+            shapes = [[6, ], [6, ], [4, ]]
+            params = []
+            for shape in shapes:
+                params.append(np.random.random(shape))
+
+        The resulting circuit in this case is:
+
+        >>> print(qml.draw(circuit)())
```
Comment (Jaybsoni) [inline pennylane/templates/subroutines/interferometer.py:179]:
```suggestion
            @qml.qnode(dev)
            def circuit(params):
                qml.Interferometer(params, wires=range(4))
                return qml.expval(qml.Identity(0))

            shapes = [[6, ], [6, ], [4, ]]
            params = []
            for shape in shapes:
                params.append(np.random.random(shape))

        Using these random parameters, the resulting circuit is:

        >>> print(qml.draw(circuit)(params))
            0: ──╭BS(0.0522, 0.0472)────────────────────╭BS(0.438, 0.222)───R(0.606)────────────────────┤ ⟨I⟩
            1: ──╰BS(0.0522, 0.0472)──╭BS(0.994, 0.59)──╰BS(0.438, 0.222)──╭BS(0.823, 0.623)──R(0.221)──┤
            2: ──╭BS(0.636, 0.298)────╰BS(0.994, 0.59)──╭BS(0.0818, 0.72)──╰BS(0.823, 0.623)──R(0.807)──┤
            3: ──╰BS(0.636, 0.298)──────────────────────╰BS(0.0818, 0.72)───R(0.854)────────────────────┤

        Using different values for optional arguments:

        .. code-block:: python

            @qml.qnode(dev)
            def circuit():
                qml.Interferometer(params, wires=range(4), mesh='triangular', beamsplitter='clements')
                return qml.expval(qml.Identity(0))

            shapes = [[6, ], [6, ], [4, ]]
            params = []
            for shape in shapes:
                params.append(np.random.random(shape))

        The resulting circuit in this case is:

        >>> print(qml.draw(circuit)(params))
```

Comment (Jaybsoni) [inline pennylane/templates/subroutines/interferometer.py:179]:
```suggestion
            @qml.qnode(dev)
            def circuit(params):
                qml.Interferometer(*params, wires=range(4))
                return qml.expval(qml.Identity(0))

            shapes = [[6, ], [6, ], [4, ]]
            params = []
            for shape in shapes:
                params.append(np.random.random(shape))

        Using these random parameters, the resulting circuit is:

        >>> print(qml.draw(circuit)(params))
            0: ──╭BS(0.0522, 0.0472)────────────────────╭BS(0.438, 0.222)───R(0.606)────────────────────┤ ⟨I⟩
            1: ──╰BS(0.0522, 0.0472)──╭BS(0.994, 0.59)──╰BS(0.438, 0.222)──╭BS(0.823, 0.623)──R(0.221)──┤
            2: ──╭BS(0.636, 0.298)────╰BS(0.994, 0.59)──╭BS(0.0818, 0.72)──╰BS(0.823, 0.623)──R(0.807)──┤
            3: ──╰BS(0.636, 0.298)──────────────────────╰BS(0.0818, 0.72)───R(0.854)────────────────────┤

        Using different values for optional arguments:

        .. code-block:: python

            @qml.qnode(dev)
            def circuit():
                qml.Interferometer(*params, wires=range(4), mesh='triangular', beamsplitter='clements')
                return qml.expval(qml.Identity(0))

            shapes = [[6, ], [6, ], [4, ]]
            params = []
            for shape in shapes:
                params.append(np.random.random(shape))

        The resulting circuit in this case is:

        >>> print(qml.draw(circuit)(params))
```

**Comment by Jaybsoni** [comment]
Hi @ankit27kh, 

Just a few small suggestions, but other then that everything looks good to go! Thanks for the contribution 👍🏼

**Review by Jaybsoni — APPROVED** [review]

**Comment by ankit27kh** [comment]
Thanks @Jaybsoni. 

> Also, Interferometer does not have a `shape` method. As it requires different parameters with different shapes, there should be this method. Should I make a PR for it?

What do you think about this?

**Comment by Jaybsoni** [comment]
No worries, 

> Thanks @Jaybsoni.
> 
> > Also, Interferometer does not have a `shape` method. As it requires different parameters with different shapes, there should be this method. Should I make a PR for it?
> 
> What do you think about this?

Yes for sure, can you first submit a GitHub issue regarding this feature request, then you can open up another PR addressing that issue. This way if for some reason you don't have time to finish the PR someone else can read the issue/feature request and pick it up. 

Cheers,

**Pull request merged**